wordpress-mobile/WordPress-Android
# Fix JP Screenshot Automation UI Tests

Fixes #16845 together with the changes from #16846.

Since it builds on top of that PR, this is targeting the branch of that one.

Changes:
- Fix mocks for jetpack screenshot tests
- Fix typo in new features dialog layout
- Fix login in Jetpack UI tests (and dismiss full screen dialog about insights)
- Fix navigation to activity log in JP screenshots test
- Fix navigation to stats in JP screenshots test
- Use recycler view to click stats and media in site menu
- Use only one mock for `/me/sites` (WP&JP) & adapt mocks to correct site id
- Add `switchToSite` in `MySitePage` and use it
- Switch to the correct site in `navigateDiscover` (since `editBlogPost` is commented)
- Refactor `goToPosts` to use recycler view item click
- Make navigation to site menu items consistent
- Use `goToStats` instead of `clickStats`, and removed `clickStats`.
- Improve code to navigate back from media in WP screenshot tests
- Various fixes to make navigation to site menu items work when Quick Actions are displayed or not
- Dismiss dialog about Insights on Stats screen

To test:
1. Checkout the PR branch locally ([Github CLI](https://cli.github.com/) command: `gh pr checkout 16877`)
2. Switch to `jetpackWasabiDebug` build flavor.
3. Run the `jPScreenshotTest` from `JPScreenshotTest` in Android Studio to check that it **succeeds**.

   ⚠️ **Note:** _When running multiple times it fails too often because the mock for activity is not always applying, which is pretty strange but I didn't have time to look deeper into the reason behind this.
This can fail in more places than `navigateActivityLog`, sometimes it loads correctly for activity log but then won't load for one of the next tests._

   <details>
   <summary>Emulator Setup Guide</summary>

     Ideally we run the tests on an emulator created by fastlane. See the notes in the issue for how this can be done:

     > Follow the guide in `SCREENSHOT_DEVICE_SETUP.md` to create a device like the ones we're using for screenshots automation (this won't be super easy as the guide is a bit outdated and if you're on an M1 Mac you have to change the generated .ini files manually to set the arm64 architecture or else the emultators won't start.
   </details>

4. Close emulator & wipe data
5. Switch to `wordpressWasabiDebug` build flavor.
6. Run the `wPScreenshotTest` from `WPScreenshotTest` to check that it **succeeds**, meaning we're regression free.

## Regression Notes
1. Potential unintended areas of impact
   - WP Screenshot tests.
   - `EditorTests`
   - `allDayStatsLoad()`

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   - Ran WP Screenshot tests and fixed the issues by using only one mock for `/me/sites`.
   - `EditorTests` are disabled.
   - `allDayStatsLoad` fails, but it also fails without the changes.

3. What automated tests I added (or what prevented me from doing so)
   - It's only automated UI tests changes 😅.

PR submission checklist:
